### PR TITLE
AM-482 : Tests for AM-457 - Fixed and added new JUnit tests

### PR DESCRIPTION
**Pull request by nitinsahu1**
JIRA link (if applicable)
Story : https://tools.hmcts.net/jira/browse/AM-457
Task : https://tools.hmcts.net/jira/browse/AM-482

Change description
Case Document AM API-Download: Initial set of unit tests for AM-482 download binary content.

Does this PR introduce a breaking change? (check one with "x")

[ ] Yes
[ x] No

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAmControllerTest.java [hunk 1]
```diff
@@ -36,52 +59,183 @@ public void shouldGetValidMetaDataResponse() {
...
+    private UUID getUuid() {
+        return UUID.fromString(MATCHED_DOCUMENT_ID);
     }
 
-    private ResponseEntity<Object> setDocumentBinaryContent() {
-        return new ResponseEntity<Object>(new Object(), HttpStatus.ACCEPTED);
+    private CaseDocumentMetadata getCaseDocumentMetadata(String docId, List<Permission> permission) {
+        Document document = Document.builder().permissions(permission).build();
+        document.setId(docId);
+        return CaseDocumentMetadata.builder().caseId("").documents(Optional.of(Arrays.asList(document))).build();
```
Comment (dharmendra8186) [inline src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAmControllerTest.java]: Pass caseId for positive scenario.

File: src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAmControllerTest.java [hunk 2]
```diff
@@ -36,52 +59,183 @@ public void shouldGetValidMetaDataResponse() {
...
+
+    private UUID getUuid() {
+        return UUID.fromString(MATCHED_DOCUMENT_ID);
     }
 
-    private ResponseEntity<Object> setDocumentBinaryContent() {
-        return new ResponseEntity<Object>(new Object(), HttpStatus.ACCEPTED);
+    private CaseDocumentMetadata getCaseDocumentMetadata(String docId, List<Permission> permission) {
+        Document document = Document.builder().permissions(permission).build();
+        document.setId(docId);
```
Comment (dharmendra8186) [inline src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAmControllerTest.java]: remove this line and set docId from builder.

File: src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAmControllerTest.java (hunk 1 above)
Comment (nitinsahu1) [reply src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAmControllerTest.java]: Done

File: src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAmControllerTest.java (hunk 2 above)
Comment (nitinsahu1) [reply src/test/java/uk/gov/hmcts/reform/ccd/document/am/controller/endpoints/CaseDocumentAmControllerTest.java]: Done

**Pull request merged**